### ISSUE TITLE
AkkaUct failing with error java.lang.ClassNotFoundException: akka.event.DefaultLoggingFilter

### ISSUE THREAD

**Issue by dhanalla** [renaissance-benchmarks/renaissance#373]
issue reproduces on Windows and Linux.

Command: java.exe -Xms12G -Xmx12G -debug -jar renaissance-jmh-0.14.1-10.jar -bm avgt -prof stack  -f 1 -t 20 -i 20 -v EXTRA AkkaUct

JDK version:  msft open jdk build jdk-17.0.6+10

Exception :
java.lang.ClassNotFoundException: akka.event.DefaultLoggingFilter
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:467)
	at akka.actor.ReflectiveDynamicAccess.$anonfun$getClassFor$1(ReflectiveDynamicAccess.scala:29)
	at scala.util.Try$.apply(Try.scala:210)
	at akka.actor.ReflectiveDynamicAccess.getClassFor(ReflectiveDynamicAccess.scala:28)
	at akka.actor.ReflectiveDynamicAccess.createInstanceFor(ReflectiveDynamicAccess.scala:47)
	at akka.actor.ActorSystemImpl.<init>(ActorSystem.scala:922)
	at akka.actor.ActorSystem$.apply(ActorSystem.scala:272)
	at akka.actor.ActorSystem$.apply(ActorSystem.scala:316)
	at akka.actor.ActorSystem$.apply(ActorSystem.scala:290)
	at edu.rice.habanero.actors.AkkaActorState$.newActorSystem(AkkaActor.scala:196)
	at edu.rice.habanero.benchmarks.uct.UctAkkaActorBenchmark$UctAkkaActorBenchmark.runIteration(UctAkkaActorBenchmark.scala:25)
	at org.renaissance.actors.AkkaUct.$anonfun$run$1(AkkaUct.scala:46)
	at scala.collection.immutable.Range.foreach$mVc$sp(Range.scala:190)
	at org.renaissance.actors.AkkaUct.run(AkkaUct.scala:45)
	at org.renaissance.jmh.JmhRenaissanceBenchmark.runOperation(JmhRenaissanceBenchmark.java:128)
	at org.renaissance.actors.jmh_generated.JmhAkkaUct_runOperation_jmhTest.runOperation_avgt_jmhStub(JmhAkkaUct_runOperation_jmhTest.java:310)
	at org.renaissance.actors.jmh_generated.JmhAkkaUct_runOperation_jmhTest.runOperation_AverageTime(JmhAkkaUct_runOperation_jmhTest.java:231)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.openjdk.jmh.runner.BenchmarkHandler$BenchmarkTask.call(BenchmarkHandler.java:475)
	at org.openjdk.jmh.runner.BenchmarkHandler$BenchmarkTask.call(BenchmarkHandler.java:458)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)

java.lang.ClassNotFoundException: akka.event.DefaultLoggingFilter
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:467)
	at akka.actor.ReflectiveDynamicAccess.$anonfun$getClassFor$1(ReflectiveDynamicAccess.scala:29)

**Comment by lbulej** [comment]
I can reproduce this on OpenJDK 11.0.18 and Renaissance 0.14.2 when using the JMH harness. Does not happen when using the Renaissance harness. At the moment, I have no idea why, so this will need some investigation.

**Comment by lbulej** [comment]
TLDR: I will submit a PR to fix the technical cause of issue, but running with `-t 20` is not necessarily a correct way to run the benchmarks under JMH (it actually applies to #374 as well).

<details>
<summary>Details</summary><br/>

When given `-t` with a value greater than `1`, JMH executes the measured operation concurrently in multiple threads using a single benchmark instance. This is not thread-safe and is not how Renaissance benchmarks are intended to be run. Arguably, this should be more explicitly documented in the `Benchmark` interface description, but it would not really help the situation here, because it is not something visible to JMH users anyway.

Even though we do provide a way to run the benchmarks under JMH, they are not necessarily compatible with all the ways JMH can run them (and were not intended to be). Unfortunately, there does not seem to be a reasonable (declarative) way to restrict JMH to a compatible subset, i.e., to limit Renaissance benchmark execution to a single JMH runner (the `@Threads` or `@GroupThreads` annotations do not apply).

Some benchmarks exercise frameworks that need to listen on specific ports and there is no simple way to make multiple instances of such benchmarks coexist in a single JVM, so even the JMH `@State` annotations with thread scope would not help. Some benchmarks need to perform a cleanup after every invocation of the measured operation, which JMH does not do when executing the operation in multiple threads.

That said, running with `-t` *might* work for some benchmarks and probably works for `akka-uct`, because the benchmark creates a new actor system in each invocation of the measured method.

The `ClassNotFound` exception stems from the following:
- Akka loads some classes (including logging filters) using reflection, and the dynamic loading setup prefers context class loader if it is set. 
- JMH does not set the context class loader in the additional benchmark method runners to that of the main thread:
  - When run with `-t 1`, in the main runner both the `Thread.currentThread().getContextClassLoader()` and the `ActorSystem.getClass().getClassLoader()` point to the same `java.net.URLClassLoader` (which knows where to load the `akka-actor` classes from).
  - When run with `-t 2` (or more), in runners other than the main the `Thread.currentThread().getContextClassLoader()` may point to `jdk.internal.loader.ClassLoaders$AppClassLoader` which has no idea where `akka-actor` classes are, but the class loader is preferred to that of `ActorSystem.getClass().getClassLoader()` (which points to the right `java.net.URLClassLoader`).

Combined, this results in Akka failing to find its own classes on other than the main JMH thread.

It is difficult to say whether JMH should set the context class loader in the runners to that of the main thread, or whether Renaissance should set the context class loader before invoking anything on the benchmark class.

For now, we can set/restore the context class loader around each call to a `Benchmark` method in the JMH wrapper. This will avoid the crash due to wrong context at the cost of very negligible overhead to each method call. Of course it does not mean that running any benchmark with `-t` is a good idea.

This does not affect the Renaissance harness at the moment, which sets the context class loader for the main thread after loading each benchmark and calls benchmark methods from that single thread. If asked to run more than one benchmark, the harness loads and executes the benchmarks one by one, so the context class loader stays valid during execution of each benchmark. We may want to eventually make the context management more robust/explicit, but there is no hurry.
</details>
